go/src/koding/klientctl/config.go

```diff
@@ -109,57 +111,85 @@ func ConfigUnset(c *cli.Context, log logging.Logger, _ string) (int, error) {
 		return 1, nil
 	}
 
-	if err := updateKonfigCache(c.Args().Get(0), nil); err != nil {
+	if err := updateKonfigCache(c.Args().Get(0), ""); err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		return 1, err
 	}
 
 	return 0, nil
 }
 
-func updateKonfigCache(key string, value interface{}) error {
+func setFlatKeyValue(m map[string]interface{}, key, value string) error {
+	keys := strings.Split(key, ".")
+	it := m
+	last := len(keys) - 1
+
+	for _, key := range keys[:last] {
+		switch v := it[key].(type) {
+		case map[string]interface{}:
+			it = v
+		case nil:
+			newV := make(map[string]interface{})
+			it[key] = newV
+			it = newV
```

Comment:
> t, after setting key map[key:map[key:val]]
> t, after setting m map[key:val]

The snippet is correct, the tree is being build from root to the leaf - it firstly points at parent, then at child.

How about: https://play.golang.org/p/PS36IzSepi